server.js

```diff
@@ -1,33 +1,117 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+
+import booksData from './data/books.json'
+
+const port = process.env.PORT || 8080;
+const app = express();
+
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Technigo Book Club')
 })
 
-// Start the server
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`)
+app.get('/library/:page', (req, res) => {
+  const page = +req.params.page - 1
+  let showPage = booksData.slice(page*20, page*20+20)
+
+  if (showPage.length < 1 ) {
+    res.send('No books to show')
+  } else {
+    res.json(showPage)
+  }
+})
+
+app.get('/title/:title', (req, res) => {
+  const title = req.params.title
+  let specificBookTitle = booksData.filter((item) => item.title.toString().toLowerCase().includes(title.toLowerCase()))
+ 
+  if (specificBookTitle.length < 1) {
+    res.send('Could not find any books with that title')
+  } else {
+    res.json(specificBookTitle)
+  }
+})
+
+app.get('/author/:author', (req, res) => {
+  const author = req.params.author
+  let booksFromAuthor = booksData.filter((item) => item.authors.toLowerCase().includes(author.toLowerCase()))
+  
+  if (booksFromAuthor.length < 1) {
+    res.send('Could not find any author by that name')
+  } else {
+    res.json(booksFromAuthor)
+  }
 })
+
+app.get('/isbn/:isbn', (req, res) => {
+  const isbn = req.params.isbn
+  let specificBook = booksData.filter((item) => item.isbn === +isbn)
+  
+  if (specificBook.length < 1) {
+    res.send('Could not find the requested book')
+  } else {
+    res.json(specificBook)
+  }
+})
+
+app.get('/isbn13/:isbn13', (req, res) => {
+  const isbn = req.params.isbn13
+  let specificBook = booksData.filter((item) => item.isbn13 === +isbn)
+  
+  res.json(specificBook)
+
+  if (specificBook.length < 1) {
+    res.send('Could not find the requested book')
+  } else {
+    res.json(specificBook)
+  }
+})
+
+app.get('/ratings/:rating/:page', (req, res) => {
```

Comment:
Nice structure, but with page numbers stored as URL query would be even nicer :)